public/app/features/explore/ResponseErrorContainer.test.tsx

```diff
@@ -0,0 +1,56 @@
+import React from 'react';
+import { configureStore } from '../../store/configureStore';
+import { ResponseErrorContainer } from './ResponseErrorContainer';
+import { Provider } from 'react-redux';
+import { render, screen } from '@testing-library/react';
+import { ExploreId } from '../../types';
+import { DataQueryError, LoadingState } from '@grafana/data';
+
+describe('ResponseErrorContainer', () => {
+  it('shows error message if it does not contain refId', async () => {
+    setup({
+      message: 'test error',
+    });
+    screen.getByText('test error');
+  });
+
+  it('shows error.data.message if error.message does not exist', async () => {
+    setup({
+      data: {
+        message: 'test error',
+      },
+    });
+    screen.getByText('test error');
```

Comment:
same as before, but like

```suggestion
    expect(screen.queryByText('test error')).toBeInTheDocument();
```

Reply:
disregard this, apparently `getBy*` is the recommended way of checking if something *is* in the document: https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-query-variants-for-anything-except-checking-for-non-existence

Reply:
I am really surprised by this example though:
`expect(screen.getByRole('alert')).toBeInTheDocument()`
if it throws the `toBeInTheDocument` is never called right? Or am I missing something? Or does this make sense just for the success case? Like:
```
const el = screen.getByRole('alert')
expect(el).toBeInTheDocument()
```
and sort of making sure `el` isn't accidentally just `el = true` or something like that?

Reply:
Explains it here actually https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#using-get-variants-as-assertions

Reply:
oh, ok, thanks for finding itout. so basically it's just to make it more explicit.
He has a very good point there IMHO.

Reply:
Yup changed it already to the recommended format.